### ISSUE TITLE
Wrong vendor/http_parser version for v0.4.1 on Windows

### ISSUE THREAD

**Issue by mamobyz** [nodegit/nodegit#596]
When installing nodegit v0.4.1 on Windows, getVendorLib() in retrieveExternalDependencies.js fails on checking the existence of http_parser v2.5.0. As I found out, the vendor/http_parser folder contains a zero-length file 2.0.0, instead of 2.5.0. Upon that, the procedure tries to download it from github.com.

Unfortunately I sit behind a company proxy server, and the lifeCycleScripts seems not able to cope with. At the end, it can't resolve the host name github.com, which leads to failure of the installation.

I tried the same on Ubuntu, http_parser/2.5.0 is there. The installation succeeds on Ubuntu.

Please help or check what leads to the 2.0.0 file on Windows!

**Comment by mamobyz** [comment]
I found a way of resort. 

First, it seems, that nodegit v0.4.1 is shipped with http_parser v2.0.0 anyway, even though v2.5.0 is required. Therefore, during the installation, the life-cycle scripts downloads the tarball of http_parser v2.5.0 from https://github.com/... as provided in package.json.

Finally, I have to set the env-variable https_proxy to call npm install nodegit, such as 
```HTTPS_PROXY=http://... npm install nodegit```

**Comment by maxkorp** [comment]
> First, it seems, that nodegit v0.4.1 is shipped with http_parser v2.0.0 anyway, even though v2.5.0 is required.
Depends on which version of node you're building against. 0.10 (more common) requires v2.0.0. 
Once the prebuilts are out, that should resolve your issues (since the http_parser source won't be needed), but to be entirely honest I'd raise an issue with your team that the firewall policies are blocking access to needed build components. I know that's not always realistic, which is one of the reasons we provide the prebuilt binaries, but that's not a nodegit bug.

**Comment by mamobyz** [comment]
@maxkorp: You are right, indeed using proxy is somehow old-fashioned nowadays. 
No, I was trying to install 0.4.1 on both Windows and Ubuntu. The difference is that the env-setting for proxy was always present, even though the install.js was started as another node process by npm. On Windows, it seems different, the env-variable https_proxy was taken from Windows registry (or set in IE options setting), which is, I guess, not visible to the node (install.js) process. However, as then I figured out, I set this env-variable in the shell directly, and start npm install, it worked then.

Thanks for your response!
